app/models/concerns/course/assessment/submission/answers_concern.rb

```diff
@@ -17,15 +17,37 @@ def from_question(question_id)
   end
 
   def create_new_answers
-    questions_to_attempt ||= assessment.questions
+    questions_to_attempt ||= assessment.questions.includes(:actable)
     new_answers = questions_to_attempt.not_answered(self).attempt(self)
+    bulk_save_new_answers(new_answers)
+  end
+
+  private
+
+  # Insert new answer records (and its actables) in bulk.
+  #
+  # @param [Array<Course::Assessment::Answer>] new_answers Array of new submission answers
+  # @raise [ActiveRecord::RecordInvalid] If the new answers cannot be saved.
+  # @return[Boolean] If new answers were created.
+  def bulk_save_new_answers(new_answers)
+    # When there are no existing answers, the first one will be the current_answer.
+    # We first filter new_record from the new_answers and assign the current answer flag
+    # below.
+    new_answers_record = new_answers.select(&:new_record?)
+    return false unless new_answers_record.present?
+
+    new_answers_record.each do |new_answer_record|
+      new_answer_record.current_answer = true
+    end
+
+    new_answers_actables = new_answers_record.map(&:actable)
+    new_answers_group_by_actables = new_answers_actables.group_by { |actable| actable.class.to_s }
 
-    new_answers.map do |answer|
-      # When there are no existing answers, the first one will be the current_answer.
-      if answer.new_record?
-        answer.current_answer = true
-        answer.save!
+    ActiveRecord::Base.transaction do
+      new_answers_group_by_actables.each_key do |key|
+        key.constantize.import! new_answers_group_by_actables[key], recursive: true
```

Comment:
The gem activerecord-import (https://github.com/zdennis/activerecord-import) enables bulk insert without skipping validation and allowing associated new records of different tables to be inserted as well.

In Rails 6, insert_all and upsert_all are added but the functionalities above are not available yet.